### PR TITLE
add example solidity code for price report

### LINKED ISSUE #462: Solidity code example for price report
The contract must have two functions.
- Report price ETH/USD from Binance to the contract state
- Read the stored price and the block the data was stored

Let @smiled0g review this.

### PR DESCRIPTION
**Pull request by prin-r**
Fix #462

### CODE REVIEW

File: bridges/evm/contracts/example/NeoTCD.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+interface Bridge {
+  struct VerifyOracleDataResult {
+    bytes data;
+    bytes32 codeHash;
+    bytes params;
+  }
+  function relayAndVerify(bytes calldata data) external returns (VerifyOracleDataResult memory result);
+}
+
+contract NeoTCD {
```
Comment (sorawit) [inline bridges/evm/contracts/example/NeoTCD.sol]: Let's call this `SimplePriceDatabase` or something more understandable that `NeoTCD`.

File: bridges/evm/contracts/example/NeoTCD.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  bytes32 public codeHash;
+  bytes public params;
+  uint256 public latestETHPrice;
+  uint256 public latestReportedBlock;
```
Comment (sorawit) [inline bridges/evm/contracts/example/NeoTCD.sol]: Should just be `lastUpdate` for the latest timestamp.

File: bridges/evm/contracts/example/NeoTCD.sol [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    bridge = Bridge(bridgeAddress);
+  }
+
+  function update(bytes memory _reportPrice) public {
+    Bridge.VerifyOracleDataResult memory result = bridge.relayAndVerify(_reportPrice);
+
+    require(result.codeHash == codeHash, "INVALID_CODEHASH");
+    require(keccak256(result.params) == keccak256(params), "INVALID_PARAMS");
+
+    uint256 _latestETHPrice;
```
Comment (sorawit) [inline bridges/evm/contracts/example/NeoTCD.sol]: Can we re-use the library we wrote that decodes to `uint64[]`?

File: bridges/evm/contracts/example/NeoTCD.sol [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    uint256 halfSize = result.data.length/2;
+    for(uint256 i = 0; i < halfSize; i++){
+      uint256 j = i + halfSize;
+      uint256 mantissa = (2**(8*(halfSize-(i+1))));
+      _latestETHPrice = _latestETHPrice + uint256(uint8(result.data[i]))*mantissa;
+      timestamp = timestamp + uint256(uint8(result.data[j]))*mantissa;
+    }
+
+    uint256 dt;
+    if (timestamp < now) {
```
Comment (sorawit) [inline bridges/evm/contracts/example/NeoTCD.sol]: Let's just save timestamp to the state directly. This can just check that timestamp is increasing.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/example/NeoTCD.sol [hunk 5]
```diff
@@ -0,0 +1,55 @@
+pragma solidity 0.5.14;
+pragma experimental ABIEncoderV2;
+
+interface Bridge {
```
Comment (taobun) [inline bridges/evm/contracts/example/NeoTCD.sol]: Import this interface from `IBridge.sol` like other example?

File: bridges/evm/contracts/example/NeoTCD.sol [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+contract NeoTCD {
+  bytes32 public codeHash;
+  bytes public params;
+  uint256 public latestETHPrice;
+  uint256 public latestReportedBlock;
+
+  Bridge public bridge;
+
+  constructor(bytes32 _codeHash , bytes memory _params, address bridgeAddress) public {
```
Comment (taobun) [inline bridges/evm/contracts/example/NeoTCD.sol]: Using `Bridge/IBridge` type instead of address?

**Review by taobun — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/example/NeoTCD.sol (hunk 6 above)
Comment (prin-r) [reply bridges/evm/contracts/example/NeoTCD.sol]: done

File: bridges/evm/contracts/example/NeoTCD.sol (hunk 5 above)
Comment (prin-r) [reply bridges/evm/contracts/example/NeoTCD.sol]: done

File: bridges/evm/contracts/example/NeoTCD.sol (hunk 4 above)
Comment (prin-r) [reply bridges/evm/contracts/example/NeoTCD.sol]: done

File: bridges/evm/contracts/example/NeoTCD.sol (hunk 3 above)
Comment (prin-r) [reply bridges/evm/contracts/example/NeoTCD.sol]: done

File: bridges/evm/contracts/example/NeoTCD.sol (hunk 2 above)
Comment (prin-r) [reply bridges/evm/contracts/example/NeoTCD.sol]: done

File: bridges/evm/contracts/example/NeoTCD.sol (hunk 1 above)
Comment (prin-r) [reply bridges/evm/contracts/example/NeoTCD.sol]: done

**Review by taobun — DISMISSED** [review]

**Review by sorawit — DISMISSED** [review]

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**